tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+      }
+
+      // TODO: Check if the fraction calculator is correct.
+      val jvmGCFraction = app.sqlTaskAggMetrics.map {
+        taskMetrics => taskMetrics.jvmGCTimeSum * 1.0 / taskMetrics.executorCpuTime
+      }
+      if ((jvmGCFraction.sum / jvmGCFraction.size) > MAX_JVM_GCTIME_FRACTION) {
+        comments :+= "Average JVM GC time is very high. " +
+          "Other Garbage Collectors can be used for better performance"
+      }
+    }
+  }
+
+  /**
+   * Recommend memory overhead as: pinnedPoolSize + (memoryOverheadFactor * executorMemory)
+   */
+  private def recommendMemoryOverhead(pinnedPoolSize: String, executorMemory: String): Long = {
+    val pinnedPoolSizeNum = convertFromHumanReadableSize(pinnedPoolSize)
+    val executorMemoryNum = convertFromHumanReadableSize(executorMemory)
+    val minMemoryOverhead = convertFromHumanReadableSize(MIN_MEMORY_OVERHEAD)
+    (pinnedPoolSizeNum + Math.max(minMemoryOverhead,
+      DEFAULT_MEMORY_OVERHEAD_FACTOR * executorMemoryNum)).toLong
+  }
+
+  private def recommendGpuProperties(bestConfig: Config, systemProps: SystemProps): Unit = {
+    if (systemProps == null || systemProps.gpu_props == null) {
+      logWarning("GPU information is not available. Cannot recommend properties.")
+      comments :+= "'spark.task.resource.gpu.amount' should be set to 1/#cores."
+    } else {
+      // Recommendation for 'spark.task.resource.gpu.amount' based on num of cpu cores and
+      // 'spark.rapids.sql.concurrentGpuTasks' based on gpu memory
+      val numGpus: Int = systemProps.gpu_props.count
+      val numCores: Int = bestConfig.get_executor_cores
+
+      val taskResourceGpu: Double = 1.0 / numCores
+      val gpuMemoryNum: Long = convertFromHumanReadableSize(systemProps.gpu_props.memory)
+      val concurrentGpuTasks: Int = Math.min(
+        gpuMemoryNum * DEFAULT_CONCURRENT_GPU_TASKS_MULTIPLIER,
+        MAX_CONCURRENT_GPU_TASKS).toInt
+
+      bestConfig.set_task_resource_gpu(taskResourceGpu)
+      bestConfig.set_concurrent_gpu_tasks(concurrentGpuTasks)
+
+      if(numCores < concurrentGpuTasks) {
+        comments :+= s"For the given GPU, number of CPU cores is very low. It should be" +
+          s" at least equal to concurrent gpu tasks i.e. $concurrentGpuTasks."
+      }
+
+      // Recommendation for 'spark.executor.memoryOverhead', 'spark.executor.memoryOverheadFactor'
+      // or 'spark.kubernetes.memoryOverheadFactor' based on cluster scheduler, spark
+      // version. See recommendMemoryOverhead() for the calculation used.
+      getSparkProperty(app, "spark.rapids.memory.pinnedPool.size") match {
+        case Some(pinnedPoolSize) =>
+          val sparkMaster = getSparkProperty(app, "spark.master")
+          if (sparkMaster.contains("k8s")) {
+            if (compareSparkVersion(app.appInfo.head.sparkVersion, "3.3.0") > 0) {
+              if (getSparkProperty(app, "spark.executor.memoryOverheadFactor").isEmpty) {
```

Comment:
we talked about always just using spark.executor.memoryOverhead, I would prefer to see that unless the user has specifically set it.  And here we should recommend it set to pinned pool + the default it would have been.  Note that if user doesn't explicitly set it, it defaults to 0.10.

Reply:
The current flow is as follows (Added the same in the documentation):
```
if (pinnedPoolSize is not set) -> set pinnedPoolSize, calculate and set memoryOverhead
else
  if using yarn
    -> if memoryOverhead is not set, calculate and set.
  if using k8s
    -> if memoryOverheadFactor is not set, add comment.
```